pkg/controllers/reallocation/utilization.go

```diff
@@ -109,12 +112,34 @@ func (u *Utilization) terminateExpired(ctx context.Context, provisioner *v1alpha
 	if err != nil {
 		return fmt.Errorf("listing underutilized nodes, %w", err)
 	}
-
 	// 2. Trigger termination workflow if past TTLAfterEmpty
 	for _, node := range nodes {
 		if utilsnode.IsPastEmptyTTL(node) {
 			logging.FromContext(ctx).Infof("Triggering termination for empty node %s", node.Name)
-			if err := u.kubeClient.Delete(ctx, node); err != nil {
+			if err := u.KubeClient.Delete(ctx, node); err != nil {
+				return fmt.Errorf("sending delete for node %s, %w", node.Name, err)
+			}
+		}
+	}
+	return nil
+}
+
+func (u *Utilization) terminateFailedToJoin(ctx context.Context, provisioner *v1alpha3.Provisioner) error {
+	// 1. Get nodes
+	nodes, err := u.getNodes(ctx, provisioner, map[string]string{})
+	if err != nil {
+		return fmt.Errorf("listing nodes, %w", err)
+	}
+	// 2. Trigger termination workflow if node has failed to become ready for 5 minutes
+	for _, node := range nodes {
+		pods, err := u.getPods(ctx, node)
+		if err != nil {
+			logging.FromContext(ctx).Debugf("Continuing after failing to get pods for node %s", node.Name)
+		}
+		// If the node failed to join and all pods are terminating, delete the pod
+		if utilsnode.FailedToJoin(node, FailedToJoinTimeout) && pod.AreAllTerminating(pods) {
+			logging.FromContext(ctx).Infof("Deleting node %s after it failed to join in 5 minutes", node.Name)
```

Comment:
Let's create parity with
`logging.FromContext(ctx).Infof("Triggering termination for empty node %s", node.Name)`

e.g.
`logging.FromContext(ctx).Infof("Triggering termination for node that failed to join %s", node.Name)`